packages/@cdktf/provider-generator/lib/get/generator/provider-generator.ts

```diff
@@ -168,20 +169,35 @@ export class TerraformProviderGenerator {
       }
       providerResource.providerVersion = providerVersion;
       files.push(this.emitResource(providerResource));
+      this.emitResourceReadme(providerResource);
     }
 
     this.emitIndexFile(name, files);
   }
 
+  private emitResourceReadme(resource: ResourceModel): void {
+    const filePath = `${resource.namespaceFolderPath}/README.md`;
+    this.code.openFile(filePath);
+    this.code.line(`# \`${resource.terraformType}\``);
+    this.code.line();
+    const type = resource.isProvider
+      ? resource.provider
+      : resource.terraformType;
+    this.code.line(
+      `Refer to the Terraform Registory for docs: [\`${type}\`](${resource.linkToDocs}).`
+    );
+    this.code.closeFile(filePath);
+  }
+
   private emitIndexFile(provider: string, files: string[]): void {
     const folder = `providers/${provider}`;
```

Comment:
I haven't worked with Windows machines in a while, so I don't know if these hardcoded `/` will break on windows?

Reply:
I think they work for Nodejs – at least we already use them that way.

Reply:
I think `require` and `import` fixes that internally, but I'm not sure if `fs` fixes the paths too. At least, I can't find any documentation around it.